src/apps/statements/service/storeStatements/updateFullActivities.ts

```diff
@@ -43,79 +92,83 @@ export interface Opts {
-    return activity.id;
-  });
-  const updates = map(groupedActivities, (matchingActivities, activityId) => {
-    const names = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.name !== undefined
-      ) ? matchingActivity.definition.name : {};
-    });
-    const descriptions = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.description !== undefined
-      ) ? matchingActivity.definition.description : {};
-    });
-    const extensions = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.extensions !== undefined
-      ) ? matchingActivity.definition.extensions : {};
-    });
-    const types = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.type !== undefined
-      ) ? matchingActivity.definition.type : undefined;
-    }).filter((value) => {
-      return value !== undefined;
-    });
-    const moreInfos = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.moreInfo !== undefined
-      ) ? matchingActivity.definition.moreInfo : undefined;
-    }).filter((value) => {
-      return value !== undefined;
-    });
-
-    return {
-      activityId,
-      name: Object.assign({}, ...names),
-      description: Object.assign({}, ...descriptions),
-      extensions: Object.assign({}, ...extensions),
-      ...(types.length > 0 ? { type: types[types.length - 1] } : {}),
-      ...(moreInfos.length > 0 ? { moreInfo: moreInfos[moreInfos.length - 1] } : {}),
-    };
-  });
-
-  await config.repo.updateFullActivities({ updates, client });
-  // tslint:disable-next-line:max-file-line-count
+  const groupedActivities = groupBy(
+    definedActivities,
+    (activity) => activity.activityId,
+  );
+
+  const fullActivities = map(
+    groupedActivities,
+    (matchingActivities, activityId): FullActivityDatabase => {
+      const names = matchingActivities.map(
+        (matchingActivity) => matchingActivity.name,
+      );
+
+      const descriptions = matchingActivities.map(
+        (matchingActivity) => matchingActivity.description,
+      );
+
+      const extensions = matchingActivities.map(
+        (matchingActivity) => matchingActivity.extensions,
+      );
+
+      const contextActivitiesForFullActivity: FullActivityContextActivities[] =
+        matchingActivities.map((matchingActivity) =>
+          mapValues(matchingActivity.context?.contextActivities, (matchingContextActivities) =>
+            matchingContextActivities !== undefined
+              // tslint:disable-next-line:max-line-length
```

Comment:
@PrinceWaune we don't use such ignore lines in other modules within the xapi-service - is this required?